lib/openstudio-standards/standards/ashrae_90_1_prm/ashrae_90_1_prm_2019/ashrae_90_1_prm_2019.SpaceType.rb

```diff
@@ -0,0 +1,396 @@
+  # but the first instance if multiple instances.  Add a new instance/definition
+  # if no instance exists. Modify the definition for the remaining instance
+  # to have the specified values. This method does not alter any
+  # loads directly assigned to spaces.  This method skips plenums.
+  #
+  # @param space_type [OpenStudio::Model::SpaceType] space type object
+  # @param set_people [Bool] if true, set the people density.
+  #   Also, assign reasonable clothing, air velocity, and work efficiency inputs
+  #   to allow reasonable thermal comfort metrics to be calculated.
+  # @param set_lights [Bool] if true, set the lighting density, lighting fraction
+  #   to return air, fraction radiant, and fraction visible.
+  # @param set_electric_equipment [Bool] if true, set the electric equipment density
+  # @param set_gas_equipment [Bool] if true, set the gas equipment density
+  # @param set_ventilation [Bool] if true, set the ventilation rates (per-person and per-area)
+  # @param set_infiltration [Bool] if true, set the infiltration rates
+  # @return [Bool] returns true if successful, false if not
+  def space_type_apply_internal_loads(space_type, set_people, set_lights, set_electric_equipment, set_gas_equipment, set_ventilation, set_infiltration)
+    # Skip plenums
+    # Check if the space type name
+    # contains the word plenum.
+    if space_type.name.get.to_s.downcase.include?('plenum')
+      return false
+    end
+
+    if space_type.standardsSpaceType.is_initialized
+      if space_type.standardsSpaceType.get.downcase.include?('plenum')
+        return false
+      end
+    end
+
+    # Pre-process the light instances in the space type
+    # Remove all instances but leave one in the space type
+    instances = space_type.lights.sort
+    if instances.size.zero?
+      definition = OpenStudio::Model::LightsDefinition.new(space_type.model)
+      definition.setName("#{space_type.name} Lights Definition")
+      instance = OpenStudio::Model::Lights.new(definition)
+      instance.setName("#{space_type.name} Lights")
+      instance.setSpaceType(space_type)
+      OpenStudio.logFree(OpenStudio::Info, 'openstudio.standards.SpaceType', "#{space_type.name} had no lights, one has been created.")
+      instances << instance
+    elsif instances.size > 1
+      instances.each_with_index do |inst, i|
+        next if i.zero?
+
+        OpenStudio.logFree(OpenStudio::Info, 'openstudio.standards.SpaceType', "Removed #{inst.name} from #{space_type.name}.")
+        inst.remove
+      end
+    end
+
+    # Get userdata from userdata_space and userdata_spacetype
+    user_spaces = @standards_data.key?('userdata_space') ? @standards_data['userdata_space'] : nil
+    user_spacetypes = @standards_data.key?('userdata_spacetype') ? @standards_data['userdata_spacetype'] : nil
+    if user_spaces && user_spaces.length >= 1
+      # call this function to enforce space-space_type one on one relationship
```

Comment:
In the space type section below, you are checking has_multi_lpd_values_user_data before proceding. Wouldn't we want to do that here as well? If the user data has wwr building types but not lighting space types, then we don't need to call space_to_space_type_apply_lighting.

Reply:
Added a function has_user_lpd_values as conditional test.
The function checks if the user space / spaceType has lighting data - if test failed, then it moves to next priority